test/functions.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { readJSON } from 'fs-extra'
+import mock from 'mock-fs'
+import { join } from 'pathe'
+import { NEXT_PLUGIN_NAME } from '../packages/runtime/src/constants'
+import { writeFunctionConfiguration } from '../packages/runtime/src/helpers/utilities/functions'
+
+describe('writeFunctionConfiguration', () => {
+  afterEach(() => {
+    mock.restore()
+  })
+
+  it('should write the configuration for a function using node modules version of @netlify/plugin-nextjs', async () => {
+    const nextRuntimeVersion = '23.4.5'
+
+    mock({
+      '.netlify/plugins/package.json': JSON.stringify({
+        name: 'test',
+        version: '1.0.0',
+        dependencies: {
+          '@netlify/plugin-nextjs': '29.3.4',
+        },
+      }),
+      'node_modules/@netlify/plugin-nextjs/package.json': JSON.stringify({
+        name: '@netlify/plugin-nextjs',
+        version: nextRuntimeVersion,
+      }),
+      '.netlify/functions/some-folder/someFunctionName/someFunctionName.json': '',
```

Comment:
This is writing a file to the mock file system. Might be better to just mock the directory?
```suggestion
      '.netlify/functions/some-folder/someFunctionName': {},
```

Also, see my comment above that I think we're adding an extra directory so we probably want this if the code is changed:

```suggestion
      '.netlify/functions/some-folder': {},
```

Reply:
Thinking about it a bit more does `fs.writeFile` write recursively? If so, do we need to mock the directory here? I would think that 'mock-fs' would allow you to write to it and then read from it. 🤔